### PR TITLE
Avoid adding a duplicate comment with CRIB env details

### PR DESCRIPTION
**Pull request by chainchad**
This avoids creating a duplicate "CRIB Environment" details PR comment when the `crib` PR label is added. It checks if a comment already exists and, if so, it skips successive (duplicate) comments.

Tested:

1. [CI run](https://github.com/smartcontractkit/chainlink/actions/runs/8141923133/job/22250311682) which added the comment with the CRIB env details.
2. [CI run](https://github.com/smartcontractkit/chainlink/actions/runs/8141956439/job/22250415932?pr=12274) which skipped adding another comment because one existed.

### CODE REVIEW

File: .github/scripts/pr-comment-crib-env.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
+const core = require("@actions/core");
+const github = require("@actions/github");
+const { Octokit } = require("@octokit/rest");
```
Comment (chainchad) [inline .github/scripts/pr-comment-crib-env.js:3]: This changed from the original GitHub Script inside the workflow

File: .github/scripts/pr-comment-crib-env.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      labelsToCheck.includes(label.name)
+    );
+
+    if (!labelMatches) {
+      core.info("No 'crib' PR label found. Proceeding.");
+      return;
+    }
+
+    // Construct the comment
+    const comment = `${commentHeader} :information_source:
```
Comment (chainchad) [inline .github/scripts/pr-comment-crib-env.js:59]: This changed from the original GitHub Script inside the workflow

File: .github/scripts/pr-comment-crib-env.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  }
+}
+
+// Run the script if it's executed directly from the command line
+if (require.main === module) {
+  run();
+}
+
+// Export the run function for testing purposes
+module.exports = { run };
```
Comment (chainchad) [inline .github/scripts/pr-comment-crib-env.js:98]: This changed from the original GitHub Script inside the workflow

File: .github/scripts/crib/pr-comment-crib-env.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+    // List labels on the PR
+    const { data: labels } = await octokit.rest.issues.listLabelsOnIssue({
+      owner,
+      repo,
+      issue_number: prNumber,
+    });
+
+    // Check if any label matches the labelsToCheck
+    const labelMatches = labels.some((label) =>
```
Comment (momentmaker) [inline .github/scripts/crib/pr-comment-crib-env.js:51]: would it make sense to check for `crib` label first before checking for duplicate comment? presumably most PRs don't have the `crib` label?

Comment (chainchad) [reply .github/scripts/crib/pr-comment-crib-env.js:42]: Great catch, updated order in 5d672779a43a888b5a2657dcd89503be509793da

**Review by momentmaker — APPROVED** [review]

**Pull request merged**